### PR TITLE
Added McCarthys conditional to Sketch.hs

### PR DESCRIPTION
**Pull request by bolt12**
Referencing #21

### CODE REVIEW

File: test/Sketch.hs [hunk 1]
```diff
@@ -444,6 +444,28 @@ t7 x y z =
...
+------------------------ McCarthy's Conditional combinator -------------------------
+-- See: http://www4.di.uminho.pt/~jno/ps/pdbc.pdf
+-- And also: https://themattchan.com/docs/algprog.pdf
+
+-- Guard function used in McCarthy's conditional
+-- | It provides information about the outcome of testing @p@ on some input @a@,
+-- encoded in terms of the coproduct injections without losing the input
+-- @a@ itself.
+grdS :: Selective f => f (a -> Bool) -> f a -> f (Either a a)
+grdS f a = (selector <$> (f <*> a)) <*> a
```
Comment (snowleopard) [inline test/Sketch.hs]: I'm confused: this function doesn't seem to use the `Selective` interface, only `Applicative`. Is the implementation wrong, or the type signature?

**Comment by snowleopard** [comment]
Thanks for moving the code. Happy to merge, but see my comment!

File: test/Sketch.hs (hunk 1 above)
Comment (bolt12) [reply test/Sketch.hs]: Yes, my bad. Thanks for noticing, the signature should use `Applicative` instead of `Selective`!

**Comment by snowleopard** [comment]
@bolt12 Thanks for the fix, merged!

**Comment by bolt12** [comment]
Thanks! 😄😄

**Pull request merged**